src/main/java/io/vertx/codegen/CodeGenProcessor.java

```diff
@@ -137,45 +147,71 @@ private static void loadJsonMappers(List<CodeGen.Converter> list, InputStream is
     });
   }
 
+  private Path fetchSourcePath() {
+    try {
+      JavaFileObject generationForPath = processingEnv.getFiler()
+          .createClassFile("PathFor" + getClass().getSimpleName());
+      return Paths.get(generationForPath.toUri().getPath()).getParent();
+    } catch (IOException e) {
+      processingEnv.getMessager().printMessage(Diagnostic.Kind.WARNING, "Unable to determine source file path!");
+    }
+
+    return Paths.get("");
+  }  
+  
   private List<CodeGen.Converter> loadJsonMappers() {
+    boolean loadOK = false;
     List<CodeGen.Converter> merged = new ArrayList<>();
     for (StandardLocation loc : StandardLocation.values()) {
       try {
-        FileObject file = processingEnv.getFiler().getResource(loc, "", "META-INF/vertx/json-mappers.properties");
+        FileObject file = processingEnv.getFiler().getResource(loc, "", JSON_MAPPERS_PROPERTIES_PATH);
         try(InputStream is = file.openInputStream()) {
           try {
             loadJsonMappers(merged, is);
+            loadOK = true;
           } catch (IOException e) {
-            log.log(Level.SEVERE, "Could not load json-mappers.properties", e);
+            log.log(Level.SEVERE, COULD_NOT_LOAD_JSON_MAPPERS_PROPERTIES, e);
```

Comment:
I think instead of `loadOK` boolean we should have instead the `IOException` retained and void logging it.

if the later block that checks `loadOk` fails then we log the exception only at this time to avoid logging the exception in Eclipse IDE.

Reply:
ok